scanner/engine/master.cpp

```diff
@@ -414,28 +565,242 @@ grpc::Status MasterImpl::NewJob(grpc::ServerContext* context,
   job_result->set_success(true);
   set_database_path(db_params_.db_path);
 
-  job_params_.CopyFrom(*job_params);
+  job_params_.Clear();
+  job_params_.MergeFrom(*job_params);
+  {
+    std::unique_lock<std::mutex> lock(finished_mutex_);
+    finished_ = false;
+  }
+  finished_cv_.notify_one();
+
+  {
```

Comment:
Yes, `finished_` indicates all tasks are finished, whereas `active_job_` indicates that the job is still running. A job can be finished_ but still active as it is waiting to clean up workers or write out any final metadata.